### PR TITLE
Test: Install always DNS deployments

### PR DESCRIPTION
**Pull request by eloycoto**
Now that CoreDNS manifest are in the provision path, the use of
INSTALL_KUBEDNS is not longer needed and should be installed always.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4862)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: test/provision/k8s_install.sh [hunk 1]
```diff
@@ -23,7 +23,6 @@ export KUBEADM_CRI_SOCKET="/var/run/dockershim.sock"
 export KUBEADM_SLAVE_OPTIONS=""
 export KUBEADM_OPTIONS=""
 export K8S_FULL_VERSION=""
-export INSTALL_KUBEDNS=1
 export DNS_DEPLOYMENT="${PROVISIONSRC}/manifest/dns_deployment.yaml"
```
Comment (aanm) [inline test/provision/k8s_install.sh:26]: When is this file created?

Comment (ianvernon) [reply test/provision/k8s_install.sh:26]: @aanm depending upon the K8s version being used a symlink is created accordingly to kube-dns or core-dns.

**Comment by eloycoto** [comment]
The issue that happened on 1.12 is a know issue, and it's going to be fixed on https://github.com/cilium/cilium/pull/4757

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
Build failure: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5268/testReport/junit/k8s-1/11/K8sValidatedServicesTest_Checks_service_across_nodes_Tests_NodePort/

Output:
```
Stacktrace

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:376
Pod "testclient-tv2zr" can not connect to service "http://192.168.36.12:32520"
Expected command: kubectl exec -n default testclient-tv2zr -- curl -s -D /dev/stderr --fail --connect-timeout 3 --max-time 3 http://192.168.36.12:32520 
To succeed, but it failed:
Exitcode: 7 
Stdout:
 	 
Stderr:
 	 command terminated with exit code 7
	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/Services.go:205
Standard Output

Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 0
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Cilium pods: [cilium-f675v cilium-pd4fm]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
     testclient-tv2zr                   =>   none
     testclient-zvqk6                   =>   none
     cilium-health-k8s2                 =>   none
     coredns-b9476b976-j2mnx            =>   none
     prometheus-core-6546477dc8-8x6js   =>   none
     testds-8rdcr                       =>   none
     testds-jxnx8                       =>   none
     cilium-health-k8s1                 =>   none
     coredns-b9476b976-6d7xd            =>   none
Cilium agent "cilium-f675v": Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 37 Failed 0
Cilium agent "cilium-pd4fm": Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 32 Failed 0

Standard Error

=== Test Finished at 2018-07-25T03:49:18Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
default       testclient-tv2zr                   1/1       Running   0          1m        10.10.0.13      k8s1
default       testclient-zvqk6                   1/1       Running   0          1m        10.10.1.140     k8s2
default       testds-8rdcr                       1/1       Running   0          1m        10.10.0.94      k8s1
default       testds-jxnx8                       1/1       Running   0          1m        10.10.1.1       k8s2
kube-system   cilium-f675v                       1/1       Running   0          3m        192.168.36.12   k8s2
kube-system   cilium-pd4fm                       1/1       Running   0          3m        192.168.36.11   k8s1
kube-system   coredns-b9476b976-6d7xd            1/1       Running   0          13m       10.10.1.32      k8s2
kube-system   coredns-b9476b976-j2mnx            1/1       Running   0          13m       10.10.0.193     k8s1
kube-system   etcd-k8s1                          1/1       Running   0          21m       192.168.36.11   k8s1
kube-system   kube-apiserver-k8s1                1/1       Running   0          20m       192.168.36.11   k8s1
kube-system   kube-controller-manager-k8s1       1/1       Running   0          20m       192.168.36.11   k8s1
kube-system   kube-proxy-7wtql                   1/1       Running   0          14m       192.168.36.12   k8s2
kube-system   kube-proxy-88lk4                   1/1       Running   0          22m       192.168.36.11   k8s1
kube-system   kube-scheduler-k8s1                1/1       Running   0          21m       192.168.36.11   k8s1
prometheus    prometheus-core-6546477dc8-8x6js   1/1       Running   0          14m       10.10.1.252     k8s2

cmd: kubectl exec -n kube-system cilium-f675v -- cilium service list
ID   Frontend             Backend                   
1    10.96.0.1:443        1 => 192.168.36.11:6443   
2    10.99.226.163:9090   1 => 10.10.1.252:9090     
3    10.96.0.10:53        1 => 10.10.0.193:53       
                          2 => 10.10.1.32:53        
5    10.108.210.231:80    1 => 10.10.0.94:80        
                          2 => 10.10.1.1:80         
6    10.96.217.19:10080   1 => 10.10.0.94:80        
                          2 => 10.10.1.1:80         

cmd: kubectl exec -n kube-system cilium-f675v -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                                
7676       Disabled           Disabled          20108      k8s:io.cilium.k8s.policy.serviceaccount=default          f00d::a0a:100:0:1dfc   10.10.1.1     ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                               
                                                           k8s:zgroup=testDS                                                                                     
13949      Disabled           Disabled          14331      k8s:io.cilium.k8s.policy.serviceaccount=coredns          f00d::a0a:100:0:367d   10.10.1.32    ready   
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                           
                                                           k8s:k8s-app=kube-dns                                                                                  
29628      Disabled           Disabled          37150      k8s:app=prometheus                                       f00d::a0a:100:0:73bc   10.10.1.252   ready   
                                                           k8s:component=core                                                                                    
                                                           k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s                                                
                                                           k8s:io.kubernetes.pod.namespace=prometheus                                                            
31955      Disabled           Disabled          10598      k8s:io.cilium.k8s.policy.serviceaccount=default          f00d::a0a:100:0:7cd3   10.10.1.140   ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                               
                                                           k8s:zgroup=testDSClient                                                                               
60263      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:eb67   10.10.1.41    ready   

cmd: kubectl exec -n kube-system cilium-pd4fm -- cilium service list
ID   Frontend             Backend                   
1    10.96.0.1:443        1 => 192.168.36.11:6443   
2    10.99.226.163:9090   1 => 10.10.1.252:9090     
3    10.96.0.10:53        1 => 10.10.0.193:53       
                          2 => 10.10.1.32:53        
5    10.108.210.231:80    1 => 10.10.0.94:80        
                          2 => 10.10.1.1:80         
6    10.96.217.19:10080   1 => 10.10.0.94:80        
                          2 => 10.10.1.1:80         

cmd: kubectl exec -n kube-system cilium-pd4fm -- cilium endpoint list
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                       IPv6                 IPv4          STATUS   
           ENFORCEMENT        ENFORCEMENT                                                                                                       
53496      Disabled           Disabled          4          reserved:health                                   f00d::a0a:0:0:d0f8   10.10.0.87    ready   
54107      Disabled           Disabled          20108      k8s:io.cilium.k8s.policy.serviceaccount=default   f00d::a0a:0:0:d35b   10.10.0.94    ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                      
                                                           k8s:zgroup=testDS                                                                            
60670      Disabled           Disabled          14331      k8s:io.cilium.k8s.policy.serviceaccount=coredns   f00d::a0a:0:0:ecfe   10.10.0.193   ready   
                                                           k8s:io.kubernetes.pod.namespace=kube-system                                                  
                                                           k8s:k8s-app=kube-dns                                                                         
61932      Disabled           Disabled          10598      k8s:io.cilium.k8s.policy.serviceaccount=default   f00d::a0a:0:0:f1ec   10.10.0.13    ready   
                                                           k8s:io.kubernetes.pod.namespace=default                                                      
                                                           k8s:zgroup=testDSClient                                               
```

Logs: 
[f0c5341e_K8sValidatedServicesTest_Checks_service_across_nodes_Tests_NodePort.zip](https://github.com/cilium/cilium/files/2228694/f0c5341e_K8sValidatedServicesTest_Checks_service_across_nodes_Tests_NodePort.zip)

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
@eloycoto K8s 1.8 tests failed with issue waiting for DNS to be ready:

```
19:53:09     kube-dns was not able to get into ready state
19:53:09     Expected
19:53:09         <*errors.errorString | 0xc420360fa0>: {
19:53:09             s: "Timeout reached: timed out waiting for pods with filter -l k8s-app=kube-dns to be ready",
19:53:09         }
19:53:09     to be nil
```

Build link: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/593/

**Comment by ianvernon** [comment]
test-missed-k8s

**Review by aanm — APPROVED** [review]

**Pull request merged**